### PR TITLE
changed corpType to TMP, placed NR# in name if present, updated error handling now that deletes are now available

### PR DESCRIPTION
**Pull request by thorwolpert**
*Issue #:* /bcgov/entity#3779

*Description of changes:*
- change corpType on affiliation to TMP, until we have lookups for all filing types
- add NR# to the name field, if an NR is provided
- add error handling for failed affiliations

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/resources/business/business.py [hunk 1]
```diff
@@ -77,5 +78,14 @@ def post():
         if not isinstance(bootstrap, RegistrationBootstrap):
             return {'error': babel('Unable to create Incorporation Filing.')}, HTTPStatus.SERVICE_UNAVAILABLE
 
-        RegistrationBootstrapService.register_bootstrap(filing_account_id, bootstrap)
+        try:
+            business_name = json_input['filing']['incorporationApplication']['nameRequest']['nrNumber']
```
Comment (severinbeauvais) [inline legal-api/src/legal_api/resources/business/business.py:82]: The namex object name this prop "nrNum".

File: legal-api/src/legal_api/services/bootstrap.py [hunk 2]
```diff
@@ -118,7 +123,7 @@ def create_affiliation(cls, account: int, business_registration: str, business_n
 
         # Create an entity record
         entity_data = json.dumps({'businessIdentifier': business_registration,
-                                  'corpTypeCode': 'NR',  # to be deprecated
+                                  'corpTypeCode': 'TMP',  # to be deprecated
```
Comment (severinbeauvais) [inline legal-api/src/legal_api/services/bootstrap.py:126]: Is comment now obsolete?

File: legal-api/src/legal_api/resources/business/business.py (hunk 1 above)
Comment (severinbeauvais) [reply legal-api/src/legal_api/resources/business/business.py:82]: Should we change our code so it's the same?

Comment (lekshmimallika-aot) [reply legal-api/src/legal_api/resources/business/business.py:82]: Right now, we pass a only three properties from the name request ( nrNumber, legalName and legalType). If we make a schema change, we have to make the corresponding changes in all other components as well like filer, reports etc,

Comment (severinbeauvais) [reply legal-api/src/legal_api/resources/business/business.py:82]: Thor said we're going with "nrNumber" everywhere.

**Pull request merged**